arcgis-ios-sdk-samples/Route and directions/Navigate route/NavigateRouteViewController.swift

```diff
@@ -159,6 +152,36 @@ class NavigateRouteViewController: UIViewController {
         }
     }
     
+    /// Set route tracker, data source and location display with a solved route result.
+    ///
+    /// - Parameter routeResult: An `AGSRouteResult` object.
+    func setNavigation(with routeResult: AGSRouteResult) {
+        // Set the route tracker
+        routeTracker = makeRouteTracker(result: routeResult)
+        
+        // Set the mock location data source.
+        let firstRoute = routeResult.routes.first!
+        directionsList = firstRoute.directionManeuvers
+        let mockDataSource = makeDataSource(route: firstRoute)
+        initialLocation = mockDataSource.locations?.first
+        // Create a route tracker location data source to snap the location display to the route.
+        let routeTrackerLocationDataSource = AGSRouteTrackerLocationDataSource(routeTracker: routeTracker, locationDataSource: mockDataSource)
+        
+        // Set location display.
+        mapView.locationDisplay.dataSource = routeTrackerLocationDataSource
+        mapView.locationDisplay.autoPanMode = .navigation
+        // If the user navigates the map view away from the location display, activate the recenter button.
+        mapView.locationDisplay.autoPanModeChangedHandler = { [weak self] _ in
```

Comment:
The documentation for this property states the following:

> The block will be invoked on the same thread on which the event occured, which could be any arbitrary thread. You need to dispatch any UI related work to the main thread.

Please update accordingly. Also, take care to repeatedly access a `weak` variable without unwrapping it first.

Reply:
I made some changes but not sure if my understanding is correct. Please take a look! Thanks.